atom/atom.go

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package atom provides simple wrappers around numerics to enforce atomic
+// access.
+package atom
```

Comment:
I would recommend against it, personally. `atomic` is way too commonly used when referencing libraries/namespaces that deal with [atomic operations](https://en.wikipedia.org/wiki/Linearizability). Admittedly I'm new to Go, but generally naming user packages after programming constructs/theory is a Bad Idea&trade;.

"Atom" is better imo, simply because the term is only really used when referring to the Atom editor or (as you mentioned) Erlang's atoms - which appear to simply be a fancy name for constants.

Plus, `atomic` means less to me (personally) as an editor related thing than `atom` does.